### PR TITLE
Add support for exposure type exists criteria

### PR DESCRIPTION
**Pull request by kiran-chaudhary**
## Status
-  Ready

## Description
Added support for filter type `exposure exists` in Exposure type.  Specify "All" when `exposure type exists` filter is to be applied. Its a case sensitive match as of now.

## Minimum version of Demisto
- [x] 5.5.0

## Does it break backward compatibility?
   - No

### CODE REVIEW

File: Packs/Code42/Integrations/Code42/Code42_test.py [hunk 1]
```diff
@@ -1792,3 +1800,27 @@ def test_fetch_incidents_fetch_limit(code42_fetch_incidents_mock):
...
+
+    assert code42_res.outputs_prefix == "Code42.SecurityData"
+    assert code42_res.outputs_key_field == "EventID"
+    assert file_res.outputs_prefix == "File"
+
+    actual_query = code42_file_events_mock.securitydata.search_file_events.call_args[0][0]
+    filter_groups = json.loads(str(actual_query))["groups"]
+
+    # Assert that the  correct query gets made
+    assert len(filter_groups) == 4
```
Comment (alanag13) [inline Packs/Code42/Integrations/Code42/Code42_test.py]: this assertion is somewhat confusing to have in this test (it's hard to understand by looking at this test why it is important there be 4 filter groups). I would copy the same set of assertions that we have in a different test to make this more clear:

```python
   filter_groups = json.loads(str(actual_query))["groups"]
    expected_query_items = [
        ("md5Checksum", "d41d8cd98f00b204e9800998ecf8427e"),
        ("osHostName", "DESKTOP-0001"),
        ("deviceUserName", "user3@example.com"),
        ("exposure", "ApplicationRead"),
    ]
    expected_file_events = json.loads(MOCK_SECURITY_EVENT_RESPONSE)["fileEvents"]

    # Assert that the  correct query gets made
    assert len(filter_groups) == len(expected_query_items)
    for i in range(0, len(filter_groups)):
        _filter = filter_groups[i]["filters"][0]
        assert _filter["term"] == expected_query_items[i][0]
        assert _filter["value"] == expected_query_items[i][1]
```

File: Packs/Code42/Integrations/Code42/Code42.py [hunk 2]
```diff
@@ -447,7 +448,8 @@ def _create_exposure_filter(exposure_arg):
     # Because the CLI can't accept lists, convert the args to a list if the type is string.
```
Comment (alanag13) [inline Packs/Code42/Integrations/Code42/Code42.py:531]: This comment looks like it was copied from CLI code, which could be a little confusing here.

**Comment by alanag13** [comment]
looks good, just want the test adjusted slightly, so I just went ahead and tweaked that.

File: Packs/Code42/Integrations/Code42/Code42.py (hunk 2 above)
Comment (alanag13) [reply Packs/Code42/Integrations/Code42/Code42.py:531]: or is this referring to the `demisto-sdk` CLI?

Comment (alanag13) [reply Packs/Code42/Integrations/Code42/Code42.py:531]: i don't see this comment in the codebase for the CLI, so nevermind, you can leave this as is.

**Review by alanag13 — APPROVED** [review]

**Review by antazoey — CHANGES_REQUESTED** [review]
This is actually kind of a breaking change, no? We are now going to get less events from the same command then before when not giving any argument. Can we achieve the same effect without any breaking changes, such as allowing the keyword "EXISTS" as an argument for exposure?

**Review by alanag13 — CHANGES_REQUESTED** [review]
@unparalleled-js  is correct, I didn't realize that this functionality was not limited to exposure events when I advised this approach. @kiran-chaudhary can we use a value of `All` instead? If they also add other value to the list besides `All`, let's just ignore them.

File: Packs/Code42/Integrations/Code42/Code42.py [hunk 3]
```diff
@@ -519,8 +519,9 @@ def _create_hash_filter(hash_arg):
 
 def _create_exposure_filter(exposure_arg):
     # Because the CLI can't accept lists, convert the args to a list if the type is string.
-    if isinstance(exposure_arg, str):
-        exposure_arg = exposure_arg.split(",")
+    exposure_arg = [arg.strip() for arg in exposure_arg.split(",")]
```
Comment (kiran-chaudhary) [inline Packs/Code42/Integrations/Code42/Code42.py:532]: Handles case when exposure contains space after ','. 

Also had to remove above condition to get the desired behaviour. Couldn't actually debug why condition wasn't getting true, refer the difference in last commit to see what I am trying to convey if it is not clear.

File: Packs/Code42/Integrations/Code42/Code42.py (hunk 2 above)
Comment (antazoey) [reply Packs/Code42/Integrations/Code42/Code42.py:531]: This is the CLI in Cortex XSOAR, like where you type commands in War Rooms

File: Packs/Code42/Integrations/Code42/Code42.py (hunk 3 above)
Comment (antazoey) [reply Packs/Code42/Integrations/Code42/Code42.py:532]: I think this is fine... but keep in mind that the CLI is not the only way to call commands, you can execute them in additional scripts as well, in which case having to pass in a string array might feel odd. It is ok though. Kind of baffling why you couldn't reach the condition... I wonder what the actual "type" is of the argument. Maybe it's like a CLIArgument class or something

File: Packs/Code42/Integrations/Code42/Code42.py [hunk 4]
```diff
@@ -519,8 +519,9 @@ def _create_hash_filter(hash_arg):
...
+    if "All" in exposure_arg:
+        return ExposureType.exists()
     return ExposureType.is_in(exposure_arg)
```
Comment (antazoey) [inline Packs/Code42/Integrations/Code42/Code42.py:535]: I think we should do I think we should do `if exposure_arg:` before this line ... It should have been there before.

Comment (kiran-chaudhary) [reply Packs/Code42/Integrations/Code42/Code42.py:535]: whether `exposure_arg` is defined or not is taken care of in `append_result` method, so if its not defined this method won't be called.

**Review by antazoey — APPROVED** [review]

**Pull request merged**